src/components/results-list/ResultsDataFetcher.tsx

```diff
@@ -0,0 +1,136 @@
+          setParsedResults(parsedResults);
+        })
+        .catch((error) => {
+          setModal({
+            message: "Something went wrong fetching observations from the FHIR server.",
+            isError: true,
+          });
+          console.error(error);
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    };
+    requestObservations();
+  }, [ctx]);
+
+  const parseResults = (entries: { [key: string]: any }[]) => {
+    // extract patients from the data
+    const patients = entries
+      .filter((entry) => {
+        return entry.fullUrl.includes("Patient");
+      })
+      .map((entry) => entry.resource as Patient);
+
+    // extract observations from the data
+    const observations = entries
+      .filter((entry) => {
+        return entry.fullUrl.includes("Observation");
+      })
+      .map((entry) => entry.resource as Observation);
+
+    const readableResults = createReadableResults(patients, observations);
+
+    return readableResults;
+  };
+
+  const createReadableResults = (patients: Patient[], observations: Observation[]) => {
+    let readableResults: parsedResultsModel = [];
+
+    // extract the required data and store in a trimmed down data structure
+    patients.forEach((patient) => {
+      if (!patient.id || !patient.name || !patient.name[0].given) return;
+
+      const patientId = patient.id;
+      const firstName = patient.name[0].given[0];
+      const lastName = patient.name[0].family;
+
+      // return observations belonging to a patient based on the patient ID
+      const patientObservations = observations.filter((observation) => {
+        if (!observation.subject?.reference) return;
+
+        const subjectIdLong = observation.subject.reference;
+
+        return subjectIdLong.includes(patientId);
+      });
+
+      if (!patientObservations || patientObservations.length === 0) return;
+
+      // extract required data from each observation
+      let trimmedObservations: trimmedObservation[] = [];
+      patientObservations.forEach((observation) => {
+        if (!observation || !observation.component) return;
+
+        const trimmedObservation = observation.component.filter((component) => {
+          const loincCode = "48004-6";
+          if (component?.code?.coding?.[0].code === loincCode) {
+            return component?.valueCodeableConcept?.coding?.[0].display;
+          }
+        });
+
+        const cDnaChange = trimmedObservation[0].valueCodeableConcept?.coding?.[0].display;
+
+        trimmedObservations = [...trimmedObservations, { cDnaChange: cDnaChange, observationId: observation.id }];
+      });
+
+      readableResults = [
+        ...readableResults,
+        { patientId: patientId, firstName: firstName, lastName: lastName, observations: trimmedObservations },
+      ];
+    });
+
+    return readableResults;
+  };
+
+  if (!parsedResults) {
+    return <div>Getting observations.</div>;
```

Comment:
is this getting observations at that point? - I would have thought that's covered in loading spinner but maybe I'm getting confused